src/Compilers/CSharp/Portable/BoundTree/BoundExpressionExtensions.cs

```diff
@@ -5,11 +5,36 @@
 using Microsoft.CodeAnalysis.CSharp.Syntax;
 using Microsoft.CodeAnalysis.PooledObjects;
 using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.CSharp
 {
     internal static partial class BoundExpressionExtensions
     {
+        /// <summary>
+        /// Only legal for expressions that can produce l-values.
+        /// </summary>
+        public static RefKind GetRefKind(this BoundExpression node)
+        {
+            switch (node.Kind)
+            {
+                case BoundKind.Local:
+                    return ((BoundLocal)node).LocalSymbol.RefKind;
+
+                case BoundKind.Parameter:
+                    return ((BoundParameter)node).ParameterSymbol.RefKind;
+
+                case BoundKind.Call:
+                    return ((BoundCall)node).Method.RefKind;
+
+                case BoundKind.PropertyAccess:
```

Comment:
Do we need to handle ternary expression here as well?

Reply:
Oh, I see. The whole point here is that we don't.

---
In reply to: [153597330](https://github.com/dotnet/roslyn/pull/23415#discussion_r153597330) [](ancestors = 153597330)